server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -190,7 +190,11 @@ public DataStream promoteDataStream() {
      * @return backing index name
      */
     public static String getDefaultBackingIndexName(String dataStreamName, long generation) {
-        return getDefaultBackingIndexName(dataStreamName, generation, System.currentTimeMillis());
+        return getDefaultBackingIndexName(dataStreamName, generation, System.currentTimeMillis(), NEW_FEATURES_VERSION);
```

Comment:
nit: maybe use `Version.CURRENT` instead of `NEW_FEATURES_VERSION`?
The outcome is the same, but in other places we do the same.